barrels/LogMonkey/Logger.mc

```diff
@@ -0,0 +1,172 @@
+using Toybox.Time.Gregorian as Greg;
+using Toybox.Lang as Lang;
+
+module LogMonkey {
+
+    //! The Logger class provides public APIs to log various types of
+    //! messages. Each Logger class is associated with a log level which
+    //! is set within the Logger.initialize() function.
+    class Logger {
+
+        private var mLogLevel, mLogStream;
+
+        //! Creates a new Logger object.
+        //! @param [Toybox::Lang::String] The log level value to log messages through this class to
+        //! @param [Toybox::Lang::Object] An object which defines a println(message) function
+        (:debug)
+        function initialize(logLevel, logStream) {
```

Comment:
Should this be documented? Are users expected to create their own loggers?

Reply:
An earlier conversation with Nick resulted in restructuring the log classes with the idea that different implementations of the class could be swapped out/configured. I wanted to explicitly document what's expected as inputs to a Logger class so in the future if there's a configuration option that external developers could use to setup their own loggers or if someone internally is going in and modifying this barrel it's documented.

Reply:
Technically, if users want to implement their own logger, it only needs to provide the three public log methods. I'm not sure why users would want to inherit from this logger. I could be wrong.

It just seems weird to mention the last parameter to users as this is really a nasty implementation detail that allows us to test the logger.

Reply:
I don't think that's the case. The last parameter is simply stating you must provide a stream object that defines a println method. If better options for writing to a file/dedicated log get added in the future then that object can be swapped out as needed. I also think the idea was that in the future there could be some sort of setup/initialize function with LogMonkey that allows the user to define the underlying output/destination of the log which would then create the local variable objects.
```
module LogMonkey {
    var Debug;
    var Warn;
    var Error;

    function setup(logStream) {
        Debug = new Logger("D", logStream);
        Warn = new Logger("W", logStream);
        Error = new Logger("E", logStream);
    }
}
```
We're just not going to include any sort of setup now because the underlying system just doesn't support a more advanced infrastructure.